### PR TITLE
Added pendingVariables to RelayContainer.

### PR DESCRIPTION
**Pull request by Globegitter**
This is still a WIP, but wanted to create the PR already, to get some eyes on it. Should be able to get it done and tested in our app by the end of the week. Ref #1049

### CODE REVIEW

**Comment by Globegitter** [comment]
Should now be ready to review :)

**Comment by Globegitter** [comment]
Updated to latest master, added docs as well as changelog entry.

File: src/container/RelayContainer.js [hunk 1]
```diff
@@ -283,10 +283,16 @@ function createContainerComponent(
...
               variables: nextVariables,
+              pendingVariables: {}
             },
           };
         } else {
-          partialState = {};
+          partialState = {
+            relayProp: {
+              ...this.state.relayProp,
+              pendingVariables: pendingVariables,
```
Comment (josephsavona) [inline src/container/RelayContainer.js]: Nit: Use abbreviated object notation here (just `pendingVariables,`)

**Comment by jardakotesovec** [comment]
Would be awesome if this PR would make it to 0.8.2.  Its difficult to avoid calling setVariables with 0.8 when request is in progress..

**Comment by Globegitter** [comment]
The one thing I was wondering with my PR in the `RelayContainer` on line 247 (https://github.com/facebook/relay/pull/1069/files#diff-320b6df8cf530a681d201c75772401eaR247) if that could ever create a race condition where the state of the `pendingVariables` is not set but the `pendingRequest` is already set, if you fire off a lot of requests at the same tame?

I could not create that situation in our app though.

**Comment by Globegitter** [comment]
Updated to latest master. There was a manual merge with the `rawVariables` change, so hope I did not mess anything up. Tests are still passing locally.

**Comment by josephsavona** [comment]
@facebook-github-bot import

**Comment by facebook-github-bot-8** [comment]
Thanks for importing. If you are an FB employee go to [Phabricator](https://our.intern.facebook.com/intern/opensource/github/pull_request/1772009093021838/int_phab) to review.

**Comment by wincent** [comment]
@Globegitter: just an update. I've rebased this one again internally and made some necessary tweaks to preserve the `rawVariables`; it should go through review and land soon™.

**Comment by Globegitter** [comment]
@wincent much appreciated.

**Pull request closed without merging**